app/src/main/java/nodomain/freeyourgadget/gadgetbridge/export/GPXExporter.java

```diff
@@ -181,7 +181,7 @@ private void exportTrackpointExtensions(XmlSerializer ser, ActivityPoint point,
         long lowestDifference = 60 * 2 * 1000; // minimum distance is 2min
         for (ActivityPoint pointItem : trackPoints) {
             int hrItem = pointItem.getHeartRate();
-            if (HeartRateUtils.isValidHeartRateValue(hrItem)) {
+            if (HeartRateUtils.getInstance().isValidHeartRateValue(hrItem)) {
```

Comment:
Same here, remember it outside the loop.

Reply:
yes this should be cleaner this way. Will rework this.